cql3/expr/expression.cc

```diff
@@ -2694,5 +2694,26 @@ adjust_for_collection_as_maps(const expression& e) {
     });
 }
 
+bool is_token_function(const function_call& fun_call) {
+    static thread_local const functions::function_name token_function_name =
+        functions::function_name::native_function("token");
+
+    // Check that function name is "token"
+    const functions::function_name& fun_name =
+        std::visit(overloaded_functor{[](const functions::function_name& fname) { return fname; },
+                                      [](const shared_ptr<functions::function>& fun) { return fun->name(); }},
+                   fun_call.func);
+
+    return fun_name.has_keyspace() ? fun_name == token_function_name : fun_name.name == token_function_name.name;
```

Comment:
The name `token` gets converted to `system.token` during parsing. The parser has a special rule just for the token() function call:
```c++
    | K_TOKEN l=tupleOfIdentifiers type=relationType t=term
        {
          $e = binary_operator(
            function_call{functions::function_name::native_function("token"), std::move(l.elements)},
            type,
            std::move(t));
        }
```

`native_function("token")` corresponds to the name `system.token`.

I'm not sure if it's possible to end up with a function name that doesn't have a keyspace, but I guess it's better to handle it just in case. I took this code from `functions::get`.